src/main/java/org/humancellatlas/ingest/project/ProjectService.java

```diff
@@ -101,14 +105,30 @@ public Project update(final Project project, ObjectNode patch, Boolean sendNotif
             project.setCataloguedDate(Instant.now());
         }
 
+        AuditLog wranglingStateUpdate = wranglingStateUpdate(project, patch);
         Project updatedProject = metadataUpdateService.update(project, patch);
 
         if (sendNotification) {
             projectEventHandler.editedProjectMetadata(updatedProject);
         }
+
+        if (wranglingStateUpdate != null) {
+            auditLogService.addAuditLog(wranglingStateUpdate);
+        }
+
         return updatedProject;
     }
 
+    private AuditLog wranglingStateUpdate(Project project, ObjectNode patch) {
+        WranglingState newWranglingState = patch.has("wranglingState") ?
+                WranglingState.getName(patch.get("wranglingState").asText()) : null;
+
+        if(project.getWranglingState() != (newWranglingState)) {
+            return new AuditLog(AuditType.STATUS_UPDATED, project.getWranglingState(), newWranglingState, project);
+        }
+
+        return null;
+}
```

Comment:
I would push this into the [metadataUpdateService.updade](https://github.com/ebi-ait/ingest-core/blob/ab6eccd763fcd55ecfddf94045533a87d5dc8aa5/src/main/java/org/humancellatlas/ingest/core/service/MetadataUpdateService.java#L37), which you are calling on line 109 here.

It's a good place to monitor all types of updates to all types of metadata, which is where we will be going with this feature long term.

Reply:
we're probably going to extend this to keeping track of submission level state changes as well -- I'm not sure how well putting this in [metadataUpdateService.update](https://github.com/ebi-ait/ingest-core/blob/ab6eccd763fcd55ecfddf94045533a87d5dc8aa5/src/main/java/org/humancellatlas/ingest/core/service/MetadataUpdateService.java#L37) would work out with that change

Reply:
Had a call with Yusra on this and we discussed the intricacies of AbstractEntity DBRefs. Happy with this implementation based on where the epic is heading.